### PR TITLE
Handle bitmap lookup for nonexistent operators

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
The `GetQuorumBitmapIndicesAtBlockNumber` will revert if any of the operator is not found, and hence fail all the operators.

This PR goes back to use single RPC call for each operator to handle this issue.

Ideally, the smart contract APIs should have better error handling, e.g. distinguishing between NOT_FOUND and real error.

Tested: ran this for the entire history since V2 contracts. It turned out not a big perf cost.

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: core/eth/tx.go [hunk 1]
```diff
@@ -577,16 +578,43 @@ func (t *Transactor) GetCurrentQuorumBitmapByOperatorId(ctx context.Context, ope
...
+	indexChan := make(chan BitmapIndexOrError, len(operatorIds))
+	indexPool := workerpool.New(maxNumWorkerPoolThreads)
+	for i, id := range operatorIds {
+		i := i
+		byteId := [32]byte(id)
+		indexPool.Submit(func() {
+			result, err := t.Bindings.RegistryCoordinator.GetQuorumBitmapIndicesAtBlockNumber(&bind.CallOpts{
+				Context: ctx,
+			}, blockNumber, [][32]byte{byteId})
+			if err != nil || len(result) != -1 {
```
Comment (bxue-l2) [inline core/eth/tx.go:598]: I don't think len can return -1, it should be 1, right?

**Review by bxue-l2 — COMMENTED** [review]
Is the root case coming from "smart contract APIs" that is part of smart contract, or subgraph indexer?

**Comment by jianoaix** [comment]
> Is the root case coming from "smart contract APIs" that is part of smart contract, or subgraph indexer? I tried to go over the nonsigner code, it would be really useful to understand if we have a doc like "assignment.md" in the future

The smart contract API: it reverts if any operatorID doesn't have a bitmap. Ideally it should return a blank bitmap.
We will make documentation based on https://docs.google.com/document/d/13nxDXTAxRhT18qZrEkTtS2F7CxjlM0qA-2EGim1bu1k/edit#heading=h.3n1iaad5dsjb

**Comment by jianoaix** [comment]
ping, this PR is 3 days

File: core/eth/tx.go [hunk 2]
```diff
@@ -616,9 +648,11 @@ func (t *Transactor) GetQuorumBitmapForOperatorsAtBlockNumber(ctx context.Contex
 	bitmaps := make([]*big.Int, len(quorumBitmapIndices))
 	for result := range resultChan {
 		if result.err != nil {
-			return nil, result.err
+			// For operators not found bitmap, set the bitmap to empty.
```
Comment (ian-shim) [inline core/eth/tx.go:651]: is it worth logging these errors?

**Review by ian-shim — APPROVED** [review]

**Pull request merged**